Bindings/Java/OpenSimJNI/Test/testContext.cpp

```diff
@@ -249,6 +252,8 @@ int main()
         // in GUI use case this gets propagated to users
         cout << "Exception: " << e.what() << endl;
     }
+    const Object& connecteeAfter = psocket.getConnecteeAsObject();
+    assert(&connecteeAfter == &connecteeBefore);
```

Comment:
@aymanhab I believe *assert* will only abort in debug.  You can throw with:
 `OPENSIM_THROW_IF(&connecteeAfter == &connecteeBefore, Exception, message)`

and remove `#include <OpenSim/Auxiliary/auxiliaryTestFunctions.h>` since you are not using any of its macros,